### PR TITLE
Extract certificate generator service

### PR DESCRIPTION
**Pull request by jfractalus**
# Summary

Extract certificate generation service to separate module

# Checklist for Definition of Done

* [x] Code is technically reviewed by myself
* [x] Code is manually tested
* [x] Automatic tests for new functionality are created
* [x] Documentation is updated
* [x] Git history is clean

### CODE REVIEW

File: service/certificate-generator/src/main/java/de/adorsys/psd2/sandbox/certificate/CertificateGeneratorBasePackage.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package de.adorsys.psd2.sandbox.certificate;
+
+public class CertificateGeneratorBasePackage {
```
Comment (DG0lden) [inline service/certificate-generator/src/main/java/de/adorsys/psd2/sandbox/certificate/CertificateGeneratorBasePackage.java:3]: `final`, private Constructor?

File: service/certificate-generator/src/main/java/de/adorsys/psd2/sandbox/certificate/domain/CertificateRequest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
```
Comment (DG0lden) [inline service/certificate-generator/src/main/java/de/adorsys/psd2/sandbox/certificate/domain/CertificateRequest.java]: Does anyone use it?
Do we need AllArgsConstructor, if class is mutable?

File: service/certificate-generator/pom.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    <groupId>de.adorsys.psd2</groupId>
+    <version>1.5.0-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>psd2-certificate-generator</artifactId>
+
+  <dependencies>
+    <!-- Spring Boot -->
+    <dependency>
```
Comment (DG0lden) [inline service/certificate-generator/pom.xml:18]: do we need all these starters here?

**Review by DG0lden — CHANGES_REQUESTED** [review]

File: service/certificate-generator/pom.xml (hunk 3 above)
Comment (jfractalus) [reply service/certificate-generator/pom.xml:18]: removed unnecessary dependencies

**Pull request merged**